generator/src/main/java/com/scottlogic/deg/generator/VisualiseExecute.java

```diff
@@ -20,55 +25,44 @@
-    parameterListHeading = "%nParameters:%n",
-    optionListHeading = "%nOptions:%n",
-    abbreviateSynopsis = true)
-public class Visualise implements Runnable {
-    @CommandLine.Parameters(index = "0", description = "The path of the profile json file.")
-    private File profileFile;
-
-    @CommandLine.Parameters(index = "1", description = "The path of the output visualise file.")
-    private Path outputPath;
-
-    @CommandLine.Option(
-        names = {"-t", "--title"},
-        description = "The title to place at the top of the file")
-    private String titleOverride;
-
-    @CommandLine.Option(
-        names = {"--no-title"},
-        description = "Hides the title from the output")
-    private boolean shouldHideTitle;
-
-    @CommandLine.Option(
-            names = {"--no-optimise"},
-            description = "Prevents tree optimisation",
-            hidden = true)
-    private boolean dontOptimise;
-
-    @CommandLine.Option(
-        names = {"--no-simplify"},
-        description = "Prevents tree simplification",
-        hidden = true)
-    private boolean dontSimplify;
-
-    @CommandLine.Option(
-        names = "--help",
-        usageHelp = true,
-        description = "Display these available command line options")
-    boolean help;
+/**
+ * This class performs the visualisation of a profile and outputs the visualisation as a graphviz dot file..
+ */
+public class VisualiseExecute implements Runnable {
+
+    private final ErrorReporter errorReporter;
+    private final OutputTarget outputTarget;
+    private final ProfileReader profileReader;
+    private final VisualisationConfigSource configSource;
+    private final VisualisationConfigValidator validator;
+
+    @Inject
+    public VisualiseExecute(OutputTarget outputTarget,
+                            ProfileReader profileReader,
+                            VisualisationConfigSource configSource,
+                            VisualisationConfigValidator validator,
+                            ErrorReporter errorReporter) {
+        this.errorReporter = errorReporter;
+        this.configSource = configSource;
+        this.outputTarget = outputTarget;
+        this.profileReader = profileReader;
+        this.validator = validator;
+    }
 
     @Override
     public void run() {
+
         final DecisionTreeFactory profileAnalyser = new ProfileDecisionTreeFactory();
-        final Profile profile;
 
+        ValidationResult validationResult = validator.validateCommandLine(configSource.overwriteOutputFiles(), outputTarget);
+        if (!validationResult.isValid()) {
+            errorReporter.display(validationResult);
+            return;
+        }
+
+        final Profile profile;
         try {
-            profile = new JsonProfileReader(new NoopProfileValidator()).read(profileFile.toPath());
+            profile = new JsonProfileReader(new NoopProfileValidator()).read(configSource.getProfileFile().toPath());
```

Comment:
this should use the dependency? 
i.e. `profileReader.read(configSource.getProfileFile().toPath())`?

Reply:
okay.